ui-speedspacechart/src/components/helpers/drawElements/reticle.ts

```diff
@@ -79,225 +70,179 @@ export const drawCursor = ({ ctx, width, height, store }: DrawFunctionParams) =>
     leftOffset +
     CURVE_MARGIN_SIDES / 2;
 
-  if (cursor.x && cursor.y) {
-    // get the nearest previous and next speed values of the curve based on pointer position
-    const { previousPosition: previousCurvePosition, nextPosition: nextCurvePosition } =
-      findPreviousAndNextPosition(speeds, cursor.x!, xPositionReference);
-
-    // get the nearest previous and next speed values of the curve based on pointer position
-    const { previousPosition: previousEcoCurvePosition, nextPosition: nextEcoCurvePosition } =
-      findPreviousAndNextPosition(ecoSpeeds, cursor.x!, xPositionReference);
-
-    let stopPosition = xPositionReference(stops[0].position.start);
-
-    const filterdStops = stops.filter(({ position }) => {
-      const actualX = xPositionReference(position.start);
-      if (actualX - stopPosition < 16 && actualX !== stopPosition) {
-        return false;
-      } else {
-        stopPosition = actualX;
-        return true;
-      }
-    });
-
-    // get the nearest previous and next stop values of the curve based on pointer position
-    const { previousPosition: previousStop, nextPosition: nextStop } = findPreviousAndNextPosition(
-      filterdStops,
-      cursor.x!,
-      xPositionReference
-    );
+  // Skip drawing the reticle if the cursor is outside the graph
+  if (cursor.x === null || cursor.y === null) {
+    return;
+  }
 
-    // Get the electrical profile name based on the position of the cursor
-    if (electricalProfiles) {
-      const electricalProfileValue = electricalProfiles.find(
-        ({ position }) =>
-          xPositionReference(position.start) <= cursor.x! &&
-          xPositionReference(position.end!) >= cursor.x!
-      )?.value.electricalProfile;
-
-      if (electricalProfileValue) {
-        electricalProfileText = electricalProfileValue;
-      }
-    }
+  let cursorPosition = getCursorPosition(cursor.x!, width, store);
```

Comment:
The `!` is not necessary here, because of the previous early escape on `cursor.x === null`. Normally, TypeScript should "understands" that here `cursor.x` is necessarily a number. This comment also applies for the next occurrences of `cursor.x!` in this function.

Reply:
Done. TypeScript is not able to detect that the value not null when used in the lambda function. Check [here](https://github.com/OpenRailAssociation/osrd-ui/pull/455/files#diff-60f8e6d4b91b5be2ea8787dbc4ae1dcb69c6d99d9f6876a6a1fb8c90833b47dbR99-R100)